src/Notes/WC_Admin_Notes_Install_Jetpack_And_WooCommerce_Services_Plugin.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * WooCommerce Admin Add Install Jetpack Plugin Note Provider.
+ *
+ * Adds a note to the merchant's inbox prompting them to install the Jetpack
+ * Plugin after it fails to install during WooCommerce setup.
+ *
+ * @package WooCommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Install_Jetpack_And_WooCommerce_Services_Plugin
+ */
+class WC_Admin_Notes_Install_Jetpack_And_WooCommerce_Services_Plugin {
+	const NOTE_NAME = 'wc-admin-install-jetpack-plugin';
+
+	/**
+	 * Possibly add the Install Jetpack note.
+	 *
+	 * @param string $slug The slug of the plugin being installed.
+	 */
+	public static function possibly_add_install_jetpack_and_woocommerce_services_note( $slug ) {
+		// Exit early if we're not installing the Jetpack or the WooCommerce Services plugins.
+		if ( 'jetpack' !== $slug && 'woocommerce-services' !== $slug ) {
```

Comment:
I actually agree with your comment:

https://github.com/woocommerce/woocommerce-admin/pull/3888#discussion_r394765799

Long-term we will be making the plugins API more generic and not onboarding specific so the solution is probably to move this check to an action:

```
do_action( 'install_plugin_error', $slug, $error, ... )
```

And hook that action:
```
add_action( 'install_plugin_error', array( $this, 'possibly_add_jetpack_wcs_error_note' );
function possibly_add_jetpack_wcs_error_note( $slug ) {
    if ( $slug === 'jetpack' || $slug === 'woocommerce-services' ) {
        WC_Admin_Notes_Install_Jetpack_And_WooCommerce_Services_Plugin ::add_install_jetpack_and_woocommerce_services_note()
    }
}
```

But in the meantime, we can just add a `@todo` note here since we don't know exactly how that API will change.

Reply:
Ok, sounds good.

A pattern I've seen in WP's hooks is a dynamic hook name, like:

```
do_action( 'install_plugin_error_' . $slug, $error, ... );
```

which mean that instead of having to implement an early return in the handler you can just make it declarative in the action registration:

```
add_action( 'install_plugin_error_jetpack', array( $this, 'add_install_jetpack_and_woocommerce_services_note' ) );
add_action( 'install_plugin_error_woocommerce-services, array( $this, 'add_install_jetpack_and_woocommerce_services_note' ) );
```